internal/service/searchindex/resource_search_index_test.go

```diff
@@ -332,76 +411,30 @@ func testAccSearchIndexConfigMapping(projectIDStr, indexName, databaseName, clus
 			collection_name  = %[5]q
 			search_analyzer  = %[6]q
 			mappings_dynamic = false
-			mappings_fields  = <<-EOF
-			{
-				"address":{
-					"type":"document",
-					"fields":{
-						"city":{
-								"type":"string",
-								"analyzer":"lucene.simple",
-								"ignoreAbove":255
-						},
-						"state":{
-								"type":"string",
-								"analyzer":"lucene.english"
-						}
-					}
-				},
-				"company":{
-					"type":"string",
-					"analyzer":"lucene.whitespace",
-					"multi":{
-						"mySecondaryAnalyzer":{
-							"type":"string",
-							"analyzer":"lucene.french"
-						}
-					}
-				},
-				"employees":{
-					"type":"string",
-					"analyzer":%[6]q
-				}
-			}
-			EOF
-			analyzers        = <<-EOF
-			[
-				{
-					"name":"index_analyzer_test_name",
-					"charFilters":[
-						 {
-								"type":"mapping",
-								"mappings":{
-									 "\\":"/"
-								}
-						 }
-					],
-					"tokenizer": {
-								"type":"nGram",
-								"minGram": 2,
-								"maxGram": 5
-					},
-					"tokenFilters":[
-						 {
-								"type":"length",
-								"min":20,
-								"max":33
-						 }
-					]
-				}
-			]
-			EOF
+			%[7]s
+			%[8]s
 		}
 
 		data "mongodbatlas_search_index" "data_index" {
 			cluster_name     = %[1]s
 			project_id       = %[2]s
 			index_id 				 = mongodbatlas_search_index.test.index_id
 		}
-	`, clusterNameStr, projectIDStr, indexName, databaseName, collectionName, searchAnalyzer)
+	`, clusterNameStr, projectIDStr, indexName, databaseName, collectionName, searchAnalyzer, analyzersTF, mappingsFieldsTF)
 }
 
-func testAccSearchIndexConfigSynonyms(projectIDStr, indexName, databaseName, clusterNameStr, clusterTerraformStr string) string {
+func configSynonyms(projectIDStr, indexName, databaseName, clusterNameStr, clusterTerraformStr string, has bool) string {
```

Comment:
nit: maybe a better name for `has`? Unless I am missing something obvious here

Reply:
if it's fine I'll rename to configWithSynonyms to be consistent with other functions